mlflow/crewai/__init__.py

```diff
@@ -0,0 +1,75 @@
+"""
+The ``mlflow.crewai`` module provides an API for tracing CrewAI AI agents.
+"""
+
+import logging
+
+from packaging.version import Version
+
+from mlflow.crewai.autolog import (
+    patched_class_call,
+)
+from mlflow.utils.annotations import experimental
+from mlflow.utils.autologging_utils import autologging_integration, safe_patch
+
+_logger = logging.getLogger(__name__)
+
+FLAVOR_NAME = "crewai"
+
+
+@experimental
+@autologging_integration(FLAVOR_NAME)
+def autolog(
+    log_traces: bool = True,
+    disable: bool = False,
+    silent: bool = False,
+):
+    """
+    Enables (or disables) and configures autologging from CrewAI to MLflow.
+    Note that asynchnorous APIs and Tool calling are not recorded now.
+
+    Args:
+        log_traces: If ``True``, traces are logged for CrewAI agents.
+            If ``False``, no traces are collected during inference. Default to ``True``.
+        disable: If ``True``, disables the CrewAI autologging. Default to ``False``.
+        silent: If ``True``, suppress all event logs and warnings from MLflow during CrewAI
+            autologging. If ``False``, show all events and warnings.
+    """
+    # TODO: Handle asynchronous tasks and crew executions
+    # TODO: Tool calling is not supported since the interface of tool in CrewAI is
+    # changing drastically. Add patching once it's stabilized
+    import crewai
+
+    try:
+        class_method_map = {
+            crewai.Crew: ["kickoff", "kickoff_for_each", "train"],
+            crewai.Agent: ["execute_task"],
+            crewai.Task: ["execute_sync"],
+            crewai.LLM: ["call"],
+            crewai.Flow: ["kickoff"],
+            crewai.agents.agent_builder.base_agent_executor_mixin.CrewAgentExecutorMixin: [
```

Comment:
Perhaps this patching is not taking effect? I've run the example but this span was not created🤔

Reply:
Interesting, it appeared in my local and the unit tests for this span have passed.
<img width="760" alt="image" src="https://github.com/user-attachments/assets/b16f7bc8-9b34-49de-a53d-ab3dfda59136">

Reply:
@TomeHirata My bad, it seems sth weird is happening on my cluster env. Tested it in the new instance and works like a charm😄